File: Haskell/beginner/1177.hs
-- https://judge.beecrowd.com/pt/problems/view/1177
-- Preenchimento de Vetor II

-- Importing necessary libraries
import Control.Monad (replicateM_)
import Text.Printf (printf)

-- Recursive function to print the vector elements
vectorDisplay :: [Int] -> Int -> IO ()
vectorDisplay [] _ = return ()
vectorDisplay (x:xs) idx = do
    printf "N[%d] = %d\n" idx x
    vectorDisplay xs (idx + 1)


main :: IO ()
main = do
    -- Read the integer T from input
    input <- getLine
    let t = read input :: Int

    -- Create a vector of 1000 elements using the formula i `mod` t
    let vetor = [i `mod` t | i <- [0..999]]

    -- Print the elements of the vector
    vectorDisplay vetor 0

-- By: Fernando Serra (Thanks, Washu-sama!)
-- https://github.com/fernandosserra

File: Python/beginner/1177.py
# https://judge.beecrowd.com/pt/problems/view/1177
# Preenchimento de Vetor II

# Reading an integer input
userInput = int(input())

# Initializing a list with 1000 elements, all set to 0
systemVector = [0] * 1000

# Filling the list with values based on the input
for i in range(1000):
    systemVector[i] = i % userInput

# Printing the list
for i in range(1000):
    print(f'N[{i}] = {systemVector[i]}')
    
    
# By: Fernando Serra
# https://github.com/fernandosserra